lib/index.js

```diff
@@ -31,12 +66,12 @@ export default class Pageres {
 	 * @api public
 	 */
 
-	src(url, sizes, options) {
+	// src(url?: string, sizes?: Array<string>, options?: Options): Pageres | Array<Src> {
+	src(url, sizes, options): Pageres | Array<Src> { // TODO: how can I make all the arguments optional or all required? the commented out line doesn't work as it complains it can also be `undefined`. Can I define a separate type definiton for each scenario?
```

Comment:
What is `Value` in this case? Any docs on it? The `Intersection` docs are very minimal.

I did some pseudocode on what I'm trying to achieve: https://github.com/sindresorhus/pageres/pull/274/commits/d032ad5aed70070e09c0c3ad850291cde6477951

It's not clear how I can track properties set on `this` and then return them.